app/cdap/components/WranglerV2/SearchResultsListItem/index.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import { Box } from '@material-ui/core';
+import styled from 'styled-components';
+import ArrowForwardIosOutlinedIcon from '@material-ui/icons/ArrowForwardIosOutlined';
+
+export interface ISearchResultListItemProps {
+  primaryText: string;
+  secondaryText: string;
+  icon: boolean;
+  onClick: () => void;
+}
+
+const SearchBarWrapper = styled(Box)`
+  display: flex;
+  margin: 0px 20px;
+  padding: 15px;
+  align-items: flex-end;
+  cursor: pointer;
+  border-bottom: 1px solid #e0e0e0;
+  '& :hover': {
+    backgroundColor: #EFF0F2,
+  },
+`;
+
+const SearchBarTextWrapper = styled(Box)`
+  display: flex;
+  flex-direction: column;
+  flex: 1 1 0%;
+`;
+
+const PrimaryTextWrapper = styled(Box)`
+  margin-bottom: 10px;
+  font-size: 16px;
+  font-weight: 600;
+  line-height: 24px;
+`;
+
+const SecondaryTextWrapper = styled(Box)`
+  font-size: 14px;
+  font-weight: 400;
+  line-height: 21px;
+`;
+
+const ArrowForwardIosOutlinedIconWrapper = styled(ArrowForwardIosOutlinedIcon)`
+  margin-bottom: 3px;
+  opacity: 0.8;
+`;
+
+export default function SearchResultsListItem({
+  primaryText,
+  secondaryText,
+  icon,
+  onClick,
+}: ISearchResultListItemProps) {
+  return (
+    <SearchBarWrapper onClick={onClick}>
+      <SearchBarTextWrapper>
+        {primaryText && <PrimaryTextWrapper>{primaryText}</PrimaryTextWrapper>}
+        <SecondaryTextWrapper>{secondaryText}</SecondaryTextWrapper>
+      </SearchBarTextWrapper>
+      {icon && <ArrowForwardIosOutlinedIconWrapper />}
```

Comment:
plz make sure the styles like height, width and margins, padding for icons is as per the design
also check padding and margins across the components to achieve perfection

Reply:
To make a common search results component for both directives and the function search, I made this component which can match both component's styles,
Yes, we need to confirm this with Vinay.

Reply:
I have confirmed with Vinay and Anurag regarding this common component and they are fine with it.